Prevent TinyMCE table block from breaking placeholders in other blocks

nylen opened the pull request:
Fixes #1795.

The TinyMCE table plugin contains this code:

```js
editor.on('SelectionChange', function (e) { 
  if (hasCellSelection) { 
    e.stopImmediatePropagation();
  } 
}, true);
```

It looks like `hasCellSelection` is left as `true` when the table block loses focus.  Fortunately this is something we can fix outside of the table plugin, by calling this function inside the plugin:

```js
function clear(force) {
  // Restore selection possibilities
  editor.getBody().style.webkitUserSelect = '';

  if (force || hasCellSelection) {
    editor.$('td[data-mce-selected],th[data-mce-selected]').removeAttr('data-mce-selected');
    hasCellSelection = false;
  }
}
```

Adding the "Needs Tests" label to indicate that this will be a good candidate for an integration test once we have a framework set up to run them.

nylen committed:
Prevent TinyMCE table block from breaking placeholders in other blocks

mtias commented on blocks/library/table/table-block.js:
```diff
@@ -52,6 +52,12 @@ export default class TableBlock extends wp.element.Component {
 
+	componentDidUpdate( prevProps ) {
+		if ( prevProps.focus && ! this.props.focus && this.editor ) {
+			this.editor.plugins.table.cellSelection.clear();
```
This is actually never firing because `this.editor` is undefined, it should be `this.state.editor`. However, this also won't catch the initial render (or when focus has not been previously set on table). Something like this works, but it runs all the time:

```js
		if ( ! this.props.focus && this.state.editor && this.state.editor.plugins.table ) {
			this.state.editor.plugins.table.cellSelection.clear();
		}
```

nylen committed:
Make the thing actually work

Last time I had the right method but wasn't calling it at the right
time.  We need to call `clearSelection` when the editor initializes if
the table block is not focused.

nylen commented:
I guess some aspect of this bug tricked me last night.  It seems to be working now.

This PR has the drawback of a `setTimeout` call.  There may be a way around this but we would need a way to add our TinyMCE `editor.on( 'init' )` callback **after** this existing callback from the table plugin:

```js
editor.on('Init', function () {
  self.cellSelection = new CellSelection(editor, function (selecting) {
    if (selecting && resizeBars) {
      resizeBars.clearBars();
    }
  });
  self.resizeBars = resizeBars;
});
```

EphoxJames commented:
I'm not certain that this should have been closed. It was linked to https://github.com/WordPress/gutenberg/pull/2143 so it was auto-closed when I closed that one however I did not do anything to the table plugin in TinyMCE so this issue may still exist? CC @nylen

nylen commented:
This PR was intended to fix #1795.  According to my testing it's now fixed.  I assumed it was because of #2143 but maybe it was something else.

If #1795 comes back, let's reopen that one, but the approach used to fix it should definitely be in the table plugin rather than here.

EphoxJames commented:
It's possible that @spocke or one of the other TinyMCE guys fixed something to prevent the bug - we are on opposite sides of the world so unless I'm following Jira closely I wouldn't know about it.
